upup/pkg/fi/cloudup/azuretasks/vmscaleset.go

```diff
@@ -73,6 +73,10 @@ type VMScaleSet struct {
 	StorageProfile *VMScaleSetStorageProfile
 	// RequirePublicIP is set to true when VMs require public IPs.
 	RequirePublicIP *bool
+	// RequireLoadBalancer is set to true when VMs are using loadbalancer.
+	RequireLoadBalancer *bool
+	// LoadBalancer is the Load Balancer object the VMs will use.
+	LoadBalancerName *string
```

Comment:
Because this name is a reference to another task that kops manages (an instance of the load balancer task) we should make this field's type the task itself, that way kops can build an accurate dependency tree when applying these changes rather than trying to add a LB to a VMSS before the LB exists.

This variable only needs the LB's `Name` field set rather than all of the properties set in `pkg/model/azuremodel/api_loadbalancer.go` Kops will use the Name field to compare against the list of tasks it knows about and will populate this variable's remaining fields automatically. Hopefully this makes sense :) 

```suggestion
	LoadBalancer *LoadBalancer
```



This means the Find function below needs to set this value as:

```go
LoadBalancer: &LoadBalancer {
  Name: "foo",
}
```

and RenderAzure() will reference `LoadBalancer.Name`

Lastly, `pkg/model/azuremodel/vmscaleset.go` will need to set the VMSS task as follows:

```go
# if IG role == master ...
t.LoadBalancer = &azuretasks.LoadBalancer {
    Name: b.NameForLoadBalancer(),
}
```

Reply:
made the changes.